test/integ/cli/test_cli_asset.py

```diff
@@ -0,0 +1,139 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+"""
+Integ tests for the CLI asset commands.
+"""
+import os
+import json
+from click.testing import CliRunner
+import pytest
+import tempfile
+import posixpath
+from deadline.job_attachments.asset_manifests.hash_algorithms import hash_file, HashAlgorithm
+
+from deadline.client.cli import main
+
+
+TEST_FILE_CONTENT = "test file content"
+TEST_SUB_DIR_FILE_CONTENT = "subdir file content"
+TEST_ROOT_DIR_FILE_CONTENT = "root file content"
+
+TEST_ROOT_FILE = "root_file.txt"
+TEST_SUB_FILE = "subdir_file.txt"
+
+TEST_ROOT_DIR = "root_dir"
+TEST_SUB_DIR_1 = "subdir1"
+TEST_SUB_DIR_2 = "subdir2"
+
+
+class TestSnapshot:
+
+    @pytest.fixture
+    def temp_dir(self):
+        with tempfile.TemporaryDirectory() as tmpdir_path:
+            yield tmpdir_path
+
+    def test_pass_valid_root_dir(self, temp_dir):
+        """
+        Snapshot with a just a valid root directory, and no other parameters
+        """
+        root_dir = os.path.join(temp_dir, TEST_ROOT_DIR)
+        os.makedirs(root_dir)
+        file_path = os.path.join(root_dir, TEST_ROOT_FILE)
+        with open(file_path, "w") as f:
+            f.write(TEST_FILE_CONTENT)
+
+        runner = CliRunner()
+        runner.invoke(main, ["asset", "snapshot", "--root-dir", root_dir])
+
+        # Check manifest file details to match correct content
+        # since manifest file name is hashed depending on source location, we have to list out manifest
+        manifest_folder_path = os.path.join(root_dir, f"{os.path.basename(root_dir)}_manifests")
+        manifest_file_name = os.listdir(manifest_folder_path)[0]
```

Comment:
This seems flaky at first glance. This test relies on there being exactly one file in the `manifest_folder_path` right? Can we add an assertion here to check that there is only exactly one file in this directory, so that if this test fails because of this, we know it was due to an incorrect assumption rather than a regression in the code?

We can probably do something like:
```
manifest_files = os.listdir(manifest_folder_path)
assert len(manifest_files) == 1, f"Expected exactly one manifest file, but got {len(manifest_files)}"
manifest_file_name = manifest_files[0]
```

Reply:
It looks like this is still unaddressed. We can fix this in a future PR but it should be quick to add in here. Consider how this test will behave if there is not exactly one file in the `manifest_folder_path` (will we be able to quickly identify the issue is due to a test assumption being false rather than a regression in the code?)